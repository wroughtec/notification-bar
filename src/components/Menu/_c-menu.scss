.c-menu {
  position: absolute;
  top: var(--navbarHeight);
  right: 1.5rem;

  width: 100vw;
  max-width: 35rem;
  margin-left: 1.5rem;

  box-shadow: 0 11px 15px -7px rgba(17, 17, 17, .2), 0 24px 38px 3px rgba(17, 17, 17, .14),
    0 9px 46px 8px rgba(17, 17, 17, .12);

  @media only screen and (min-width: 35em) {
    width: 80vw;
  }

  &::after {
    position: absolute;
    top: -1rem;
    right: .5rem;

    display: block;

    content: '';

    border-color: transparent;
    border-style: solid;
    border-width: 1rem;

    border-top: 0;
    border-bottom-color: #fff;
  }

  &__list {
    padding-left: 0;
    margin-top: 0;
    margin-bottom: 0;

    list-style: none;
  }

  &__item {
    padding: .75rem;

    border-bottom: var(--menuItemBorder);

    &:last-of-type {
      border-bottom: 0;
    }
  }
}
