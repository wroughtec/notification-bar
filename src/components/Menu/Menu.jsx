// @flow
import React from 'react';
import { likeType, commentType } from 'consts/notificationTypes';
import { MenuItem } from 'components/MenuItem/MenuItem';
import './_c-menu.scss';

type Props = {
  notifications: NotificationType[]
};

export const Menu = ({ notifications }: Props) => {
  let list = <li className="c-menu__item">No new notifications</li>;

  if (notifications && notifications.length) {
    list = notifications.map(notification => {
      const {
          type,
          post: { title, id },
          likes,
          comments
        } = notification,
        key = `${type}${id}`;
      let item = null;

      switch (type) {
        case likeType:
          item = <MenuItem users={likes} title={title} type={type} />;
          break;
        case commentType:
          item = <MenuItem users={comments} title={title} type={type} />;
          break;
        default:
          item = null;
      }

      return (
        <li key={key} className="c-menu__item">
          {item}
        </li>
      );
    });
  }

  return (
    <aside className="c-menu">
      <ul className="c-menu__list">{list}</ul>
    </aside>
  );
};
