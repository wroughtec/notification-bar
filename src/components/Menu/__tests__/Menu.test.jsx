import fetchMock from 'fetch-mock';
import React from 'react';
import { shallow } from 'enzyme';
import { Menu } from 'components/Menu/Menu';
import mockData from 'mocks/test-mocky.json';

describe('Menu', () => {
  it('should render correctly with like data', () => {
    const props = { notifications: [mockData[0]] };
    const component = shallow(<Menu {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with comment data', () => {
    const props = { notifications: [mockData[1]] };
    const component = shallow(<Menu {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should not render with unknown type', () => {
    const props = {
      notifications: [
        {
          type: 'FAKE',
          post: {}
        }
      ]
    };
    const component = shallow(<Menu {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly without data', () => {
    const props = { notifications: [] };
    const component = shallow(<Menu {...props} />);
    expect(component).toMatchSnapshot();
  });
});
