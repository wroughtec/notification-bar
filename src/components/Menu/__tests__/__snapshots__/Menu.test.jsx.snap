// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should not render with unknown type 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu
    notifications={
      Array [
        Object {
          "post": Object {},
          "type": "FAKE",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="c-menu__list"
      >
        <li
          className="c-menu__item"
        />
      </ul>,
      "className": "c-menu",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="c-menu__item"
          />,
        ],
        "className": "c-menu__list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "FAKEundefined",
          "nodeType": "host",
          "props": Object {
            "children": null,
            "className": "c-menu__item",
          },
          "ref": null,
          "rendered": null,
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="c-menu__list"
        >
          <li
            className="c-menu__item"
          />
        </ul>,
        "className": "c-menu",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="c-menu__item"
            />,
          ],
          "className": "c-menu__list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "FAKEundefined",
            "nodeType": "host",
            "props": Object {
              "children": null,
              "className": "c-menu__item",
            },
            "ref": null,
            "rendered": null,
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Menu should render correctly with comment data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu
    notifications={
      Array [
        Object {
          "comments": Array [
            Object {
              "commentText": "Acme remains one of my fave company ever! The way they scale is so dynamic that makes HTML5 look static!",
              "id": "5497afbf9df3f6ff6f9ba11cdef5310f",
              "name": "Suoma Narjus",
            },
          ],
          "post": Object {
            "id": "b1638f970c3ddd528671df76c4dcf13e",
            "title": "Acme Inc dynamically scales niches worldwide",
          },
          "type": "Comment",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="c-menu__list"
      >
        <li
          className="c-menu__item"
        >
          <MenuItem
            title="Acme Inc dynamically scales niches worldwide"
            type="Comment"
            users={
              Array [
                Object {
                  "commentText": "Acme remains one of my fave company ever! The way they scale is so dynamic that makes HTML5 look static!",
                  "id": "5497afbf9df3f6ff6f9ba11cdef5310f",
                  "name": "Suoma Narjus",
                },
              ]
            }
          />
        </li>
      </ul>,
      "className": "c-menu",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="c-menu__item"
          >
            <MenuItem
              title="Acme Inc dynamically scales niches worldwide"
              type="Comment"
              users={
                Array [
                  Object {
                    "commentText": "Acme remains one of my fave company ever! The way they scale is so dynamic that makes HTML5 look static!",
                    "id": "5497afbf9df3f6ff6f9ba11cdef5310f",
                    "name": "Suoma Narjus",
                  },
                ]
              }
            />
          </li>,
        ],
        "className": "c-menu__list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Commentb1638f970c3ddd528671df76c4dcf13e",
          "nodeType": "host",
          "props": Object {
            "children": <MenuItem
              title="Acme Inc dynamically scales niches worldwide"
              type="Comment"
              users={
                Array [
                  Object {
                    "commentText": "Acme remains one of my fave company ever! The way they scale is so dynamic that makes HTML5 look static!",
                    "id": "5497afbf9df3f6ff6f9ba11cdef5310f",
                    "name": "Suoma Narjus",
                  },
                ]
              }
            />,
            "className": "c-menu__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "title": "Acme Inc dynamically scales niches worldwide",
              "type": "Comment",
              "users": Array [
                Object {
                  "commentText": "Acme remains one of my fave company ever! The way they scale is so dynamic that makes HTML5 look static!",
                  "id": "5497afbf9df3f6ff6f9ba11cdef5310f",
                  "name": "Suoma Narjus",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="c-menu__list"
        >
          <li
            className="c-menu__item"
          >
            <MenuItem
              title="Acme Inc dynamically scales niches worldwide"
              type="Comment"
              users={
                Array [
                  Object {
                    "commentText": "Acme remains one of my fave company ever! The way they scale is so dynamic that makes HTML5 look static!",
                    "id": "5497afbf9df3f6ff6f9ba11cdef5310f",
                    "name": "Suoma Narjus",
                  },
                ]
              }
            />
          </li>
        </ul>,
        "className": "c-menu",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="c-menu__item"
            >
              <MenuItem
                title="Acme Inc dynamically scales niches worldwide"
                type="Comment"
                users={
                  Array [
                    Object {
                      "commentText": "Acme remains one of my fave company ever! The way they scale is so dynamic that makes HTML5 look static!",
                      "id": "5497afbf9df3f6ff6f9ba11cdef5310f",
                      "name": "Suoma Narjus",
                    },
                  ]
                }
              />
            </li>,
          ],
          "className": "c-menu__list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Commentb1638f970c3ddd528671df76c4dcf13e",
            "nodeType": "host",
            "props": Object {
              "children": <MenuItem
                title="Acme Inc dynamically scales niches worldwide"
                type="Comment"
                users={
                  Array [
                    Object {
                      "commentText": "Acme remains one of my fave company ever! The way they scale is so dynamic that makes HTML5 look static!",
                      "id": "5497afbf9df3f6ff6f9ba11cdef5310f",
                      "name": "Suoma Narjus",
                    },
                  ]
                }
              />,
              "className": "c-menu__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "title": "Acme Inc dynamically scales niches worldwide",
                "type": "Comment",
                "users": Array [
                  Object {
                    "commentText": "Acme remains one of my fave company ever! The way they scale is so dynamic that makes HTML5 look static!",
                    "id": "5497afbf9df3f6ff6f9ba11cdef5310f",
                    "name": "Suoma Narjus",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Menu should render correctly with like data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu
    notifications={
      Array [
        Object {
          "likes": Array [
            Object {
              "id": "403f220c3d413fe9cb0b36142ebfb35d",
              "name": "Mary T. Price",
            },
            Object {
              "id": "5497afbf9df3f6ff6f9ba11cdef5310f",
              "name": "Suoma Narjus",
            },
            Object {
              "id": "fa527981cbbcb070be95854985c3188f",
              "name": "Sandra Ortega",
            },
            Object {
              "id": "4c18d43d4deccbac21a26c55f1033f53",
              "name": "William Hunt",
            },
            Object {
              "id": "38be3079117301f2f61264d6e0fbf7db",
              "name": "An Mao",
            },
          ],
          "post": Object {
            "id": "b1638f970c3ddd528671df76c4dcf13e",
            "title": "Acme Inc dynamically scales niches worldwide",
          },
          "type": "Like",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="c-menu__list"
      >
        <li
          className="c-menu__item"
        >
          <MenuItem
            title="Acme Inc dynamically scales niches worldwide"
            type="Like"
            users={
              Array [
                Object {
                  "id": "403f220c3d413fe9cb0b36142ebfb35d",
                  "name": "Mary T. Price",
                },
                Object {
                  "id": "5497afbf9df3f6ff6f9ba11cdef5310f",
                  "name": "Suoma Narjus",
                },
                Object {
                  "id": "fa527981cbbcb070be95854985c3188f",
                  "name": "Sandra Ortega",
                },
                Object {
                  "id": "4c18d43d4deccbac21a26c55f1033f53",
                  "name": "William Hunt",
                },
                Object {
                  "id": "38be3079117301f2f61264d6e0fbf7db",
                  "name": "An Mao",
                },
              ]
            }
          />
        </li>
      </ul>,
      "className": "c-menu",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="c-menu__item"
          >
            <MenuItem
              title="Acme Inc dynamically scales niches worldwide"
              type="Like"
              users={
                Array [
                  Object {
                    "id": "403f220c3d413fe9cb0b36142ebfb35d",
                    "name": "Mary T. Price",
                  },
                  Object {
                    "id": "5497afbf9df3f6ff6f9ba11cdef5310f",
                    "name": "Suoma Narjus",
                  },
                  Object {
                    "id": "fa527981cbbcb070be95854985c3188f",
                    "name": "Sandra Ortega",
                  },
                  Object {
                    "id": "4c18d43d4deccbac21a26c55f1033f53",
                    "name": "William Hunt",
                  },
                  Object {
                    "id": "38be3079117301f2f61264d6e0fbf7db",
                    "name": "An Mao",
                  },
                ]
              }
            />
          </li>,
        ],
        "className": "c-menu__list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Likeb1638f970c3ddd528671df76c4dcf13e",
          "nodeType": "host",
          "props": Object {
            "children": <MenuItem
              title="Acme Inc dynamically scales niches worldwide"
              type="Like"
              users={
                Array [
                  Object {
                    "id": "403f220c3d413fe9cb0b36142ebfb35d",
                    "name": "Mary T. Price",
                  },
                  Object {
                    "id": "5497afbf9df3f6ff6f9ba11cdef5310f",
                    "name": "Suoma Narjus",
                  },
                  Object {
                    "id": "fa527981cbbcb070be95854985c3188f",
                    "name": "Sandra Ortega",
                  },
                  Object {
                    "id": "4c18d43d4deccbac21a26c55f1033f53",
                    "name": "William Hunt",
                  },
                  Object {
                    "id": "38be3079117301f2f61264d6e0fbf7db",
                    "name": "An Mao",
                  },
                ]
              }
            />,
            "className": "c-menu__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "title": "Acme Inc dynamically scales niches worldwide",
              "type": "Like",
              "users": Array [
                Object {
                  "id": "403f220c3d413fe9cb0b36142ebfb35d",
                  "name": "Mary T. Price",
                },
                Object {
                  "id": "5497afbf9df3f6ff6f9ba11cdef5310f",
                  "name": "Suoma Narjus",
                },
                Object {
                  "id": "fa527981cbbcb070be95854985c3188f",
                  "name": "Sandra Ortega",
                },
                Object {
                  "id": "4c18d43d4deccbac21a26c55f1033f53",
                  "name": "William Hunt",
                },
                Object {
                  "id": "38be3079117301f2f61264d6e0fbf7db",
                  "name": "An Mao",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="c-menu__list"
        >
          <li
            className="c-menu__item"
          >
            <MenuItem
              title="Acme Inc dynamically scales niches worldwide"
              type="Like"
              users={
                Array [
                  Object {
                    "id": "403f220c3d413fe9cb0b36142ebfb35d",
                    "name": "Mary T. Price",
                  },
                  Object {
                    "id": "5497afbf9df3f6ff6f9ba11cdef5310f",
                    "name": "Suoma Narjus",
                  },
                  Object {
                    "id": "fa527981cbbcb070be95854985c3188f",
                    "name": "Sandra Ortega",
                  },
                  Object {
                    "id": "4c18d43d4deccbac21a26c55f1033f53",
                    "name": "William Hunt",
                  },
                  Object {
                    "id": "38be3079117301f2f61264d6e0fbf7db",
                    "name": "An Mao",
                  },
                ]
              }
            />
          </li>
        </ul>,
        "className": "c-menu",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="c-menu__item"
            >
              <MenuItem
                title="Acme Inc dynamically scales niches worldwide"
                type="Like"
                users={
                  Array [
                    Object {
                      "id": "403f220c3d413fe9cb0b36142ebfb35d",
                      "name": "Mary T. Price",
                    },
                    Object {
                      "id": "5497afbf9df3f6ff6f9ba11cdef5310f",
                      "name": "Suoma Narjus",
                    },
                    Object {
                      "id": "fa527981cbbcb070be95854985c3188f",
                      "name": "Sandra Ortega",
                    },
                    Object {
                      "id": "4c18d43d4deccbac21a26c55f1033f53",
                      "name": "William Hunt",
                    },
                    Object {
                      "id": "38be3079117301f2f61264d6e0fbf7db",
                      "name": "An Mao",
                    },
                  ]
                }
              />
            </li>,
          ],
          "className": "c-menu__list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Likeb1638f970c3ddd528671df76c4dcf13e",
            "nodeType": "host",
            "props": Object {
              "children": <MenuItem
                title="Acme Inc dynamically scales niches worldwide"
                type="Like"
                users={
                  Array [
                    Object {
                      "id": "403f220c3d413fe9cb0b36142ebfb35d",
                      "name": "Mary T. Price",
                    },
                    Object {
                      "id": "5497afbf9df3f6ff6f9ba11cdef5310f",
                      "name": "Suoma Narjus",
                    },
                    Object {
                      "id": "fa527981cbbcb070be95854985c3188f",
                      "name": "Sandra Ortega",
                    },
                    Object {
                      "id": "4c18d43d4deccbac21a26c55f1033f53",
                      "name": "William Hunt",
                    },
                    Object {
                      "id": "38be3079117301f2f61264d6e0fbf7db",
                      "name": "An Mao",
                    },
                  ]
                }
              />,
              "className": "c-menu__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "title": "Acme Inc dynamically scales niches worldwide",
                "type": "Like",
                "users": Array [
                  Object {
                    "id": "403f220c3d413fe9cb0b36142ebfb35d",
                    "name": "Mary T. Price",
                  },
                  Object {
                    "id": "5497afbf9df3f6ff6f9ba11cdef5310f",
                    "name": "Suoma Narjus",
                  },
                  Object {
                    "id": "fa527981cbbcb070be95854985c3188f",
                    "name": "Sandra Ortega",
                  },
                  Object {
                    "id": "4c18d43d4deccbac21a26c55f1033f53",
                    "name": "William Hunt",
                  },
                  Object {
                    "id": "38be3079117301f2f61264d6e0fbf7db",
                    "name": "An Mao",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Menu should render correctly without data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu
    notifications={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="c-menu__list"
      >
        <li
          className="c-menu__item"
        >
          No new notifications
        </li>
      </ul>,
      "className": "c-menu",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <li
          className="c-menu__item"
        >
          No new notifications
        </li>,
        "className": "c-menu__list",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No new notifications",
          "className": "c-menu__item",
        },
        "ref": null,
        "rendered": "No new notifications",
        "type": "li",
      },
      "type": "ul",
    },
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="c-menu__list"
        >
          <li
            className="c-menu__item"
          >
            No new notifications
          </li>
        </ul>,
        "className": "c-menu",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <li
            className="c-menu__item"
          >
            No new notifications
          </li>,
          "className": "c-menu__list",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No new notifications",
            "className": "c-menu__item",
          },
          "ref": null,
          "rendered": "No new notifications",
          "type": "li",
        },
        "type": "ul",
      },
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
